examples/default.nix

```diff
@@ -0,0 +1,42 @@
+{
+  craneLib,
+  stdenv,
+  lib,
+  hax,
+  fstar,
+  hacl-star,
+}: let
+  commonArgs = {
+    version = "0.0.1";
+    src = lib.cleanSourceWith {
+      src = craneLib.path ./..;
+      filter = path: type:
+        !builtins.isNull (builtins.match ".*(Makefile|.*[.](rs|toml|lock|diff))$" path)
+        || ("directory" == type);
+    };
+    doCheck = false;
+    cargoVendorDir = craneLib.vendorMultipleCargoDeps {
+      cargoLockList = [
+        ./Cargo.lock
+        ../Cargo.lock
+      ];
+    };
+  };
+  cargoArtifacts = craneLib.buildDepsOnly commonArgs;
+in
+  craneLib.mkCargoDerivation (commonArgs
+    // {
+      inherit cargoArtifacts;
+      pname = "hax-examples";
+      doCheck = false;
+      buildPhaseCargoCommand = ''
+        cd examples
+        export CACHE_DIR=$(mktemp -d)
+        export HINT_DIR=$(mktemp -d)
+        export SHELL=${stdenv.shell}
+        make
```

Comment:
You have to clean first, otherwise this isn't doing much intereting.

Reply:
Well, actually the sources are filtered above, so that the build is independent from generated F* or Coq (or whatever) files.
I added a comment about that, but also a `make clean` (which should be a no-op, but that doesn't hurt)